frontend/src/modules/dataset/dataset-browser.reducer.tsx

```diff
@@ -173,25 +173,25 @@ export const getDatasetContents = createAsyncThunk(
                 scope = username;
         }
 
-        const requestUrl = new URL(`${window.env.LAMBDA_ENDPOINT}dataset/${scope}/${datasetContext.name}/files`);
+        const searchParams = new URLSearchParams();
 
         if (props.nextToken) {
-            requestUrl.searchParams.set('nextToken', props.nextToken);
+            searchParams.set('nextToken', props.nextToken);
         }
 
         if (props.pageSize) {
-            requestUrl.searchParams.set('pageSize', String(props.pageSize));
+            searchParams.set('pageSize', String(props.pageSize));
         }
 
         if (props.projectName) {
-            requestUrl.searchParams.set('projectName', props.projectName);
+            searchParams.set('projectName', props.projectName);
         }
 
         if (datasetContext?.location) {
-            requestUrl.searchParams.set('prefix', datasetContext.location);
+            searchParams.set('prefix', datasetContext.location);
         }
 
-        return axios.get<ListFilesResponse>(requestUrl.toString()).then((response) => {
+        return axios.get<ListFilesResponse>(`/dataset/${scope}/${datasetContext.name}/files?${searchParams.toString()}`).then((response) => {
```

Comment:
Is this going to be a problem when running the UI locally?

Reply:
I don't think so. I thought it was which is why I originally was making the URL using the `window.env.LAMBDA_ENDPOINT` variable but looking at the rest of our project [it](https://github.com/awslabs/mlspace/blob/main/frontend/src/entities/emr/emr.reducer.ts#L54) [really](https://github.com/awslabs/mlspace/blob/main/frontend/src/entities/notebook/notebook.reducer.ts#L55) [varies](https://github.com/awslabs/mlspace/blob/main/frontend/src/entities/user/user.reducer.ts#L52) [what](https://github.com/awslabs/mlspace/blob/main/frontend/src/entities/jobs/training/training-job.reducer.tsx#L65) we use. 

I think Axios must be handling the discrepancy for us when you give it paths which is why all the other examples still work but it failed when I provided a full URL.

Even in the right next to each other in the [same file](https://github.com/awslabs/mlspace/blob/main/frontend/src/entities/user/user.reducer.ts#L57-L71) we do it different ways.

```typescript
export const addUsersToProject = createAsyncThunk(
    'user/add_users_to_project',
    async (data: UserProjectData) => {
        const requestUrl = `project/${data.projectName}/users`;
        return axios.post<IUser>(requestUrl, { usernames: data.usernames });
    }
);

export const removeUserFromProject = createAsyncThunk(
    'user/remove_user_from_project',
    async (data: IProjectUser) => {
        const requestUrl = `/project/${data.project}/users/${data.user}`;
        return axios.delete(requestUrl);
    }
);
```

Reply:
Ah yeah we've got it in the [base axios config](https://github.com/awslabs/mlspace/blob/main/frontend/src/shared/util/axios-utils.ts#L52). And then axios is handling the `/` although it would be nice if we were consistent based on the examples you linked. Either way unrelated to this so nothing to change.